packages/application-shell/src/utils/http-client.ts

```diff
@@ -84,21 +120,50 @@ const defaultUserAgent = createHttpUserAgent({
   libraryName: window.app.applicationName,
 });
 
+const defaultForwardToVersion: TForwardToConfigVersion = 'v2';
+const defaultForwardToAudiencePolicy: TForwardToAudiencePolicy =
+  'forward-url-full-path';
+
 function buildApiUrl(endpoint: string) {
   const apiUrl = getMcApiUrl().replace(/\/$/, '');
   return `${apiUrl}${endpoint}`;
 }
 
+const getAppliedForwardToHeaders = (
+  forwardToConfig?: TForwardToConfig
+): THeaders => {
+  if (!forwardToConfig) {
+    return {};
+  }
+  return {
+    ...Object.entries(forwardToConfig.headers ?? {}).reduce(
+      (customForwardHeaders, [headerName, headerValue]) => ({
+        ...customForwardHeaders,
+        // Prefix headers so that the MC API can allow and forward them.
+        [`x-forward-header-${headerName}`]: headerValue,
+      }),
+      {}
+    ),
+    [SUPPORTED_HEADERS.ACCEPT_VERSION]:
+      forwardToConfig.version ?? defaultForwardToVersion,
+    [SUPPORTED_HEADERS.X_FORWARD_TO]: forwardToConfig.uri,
+    [SUPPORTED_HEADERS.X_FORWARD_TO_AUDIENCE_POLICY]:
+      forwardToConfig.audiencePolicy ?? defaultForwardToAudiencePolicy,
```

Comment:
This is where we check if there was a passed audiencePolicy in the forwardToConfig or set to default (forward-url-full-path). Cool.